earth2studio/data/utils.py

```diff
@@ -69,6 +74,65 @@ def fetch_data(
 
     return prep_data_array(xr.concat(da, "lead_time"), device=device)
 
+def build_dataset(
+    source: DataSource,
+    time: TimeArray,
+    variable: VariableArray,
+    dataset: zarr.Array,
+    transform: None = None,
+    apache_beam_options: None = None,
```

Comment:
This should be typed to `PipelineOptions | None` right? If its pipeline options then this should be named pipeline options probably